crates/rust-analyzer/src/handlers.rs

```diff
@@ -432,9 +432,28 @@ pub(crate) fn handle_will_rename_files(
             // Limit to single-level moves for now.
             match (from_path.parent(), to_path.parent()) {
                 (Some(p1), Some(p2)) if p1 == p2 => {
-                    let new_name = to_path.file_stem()?;
-                    let new_name = new_name.to_str()?;
-                    Some((snap.url_to_file_id(&from).ok()?, new_name.to_string()))
+                    if from_path.is_dir() {
+                        // add '/' to end of url -- from `file://path/to/folder` to `file://path/to/folder/`
+                        let mut old_folder_name = from_path.file_stem()?.to_str()?.to_string();
+                        old_folder_name.push('/');
+                        let from_with_trailing_slash = from.join(&old_folder_name).ok()?;
+
+                        let imitate_from_url = from_with_trailing_slash.join("mod.rs").ok()?;
+                        let new_file_name = to_path.file_name()?.to_str()?;
+                        Some((
+                            snap.url_to_file_id(&imitate_from_url).ok()?,
+                            new_file_name.to_string(),
+                        ))
+                    } else {
+                        let old_name = from_path.file_stem()?.to_str()?;
+                        let new_name = to_path.file_stem()?.to_str()?;
+                        match (old_name, new_name) {
+                            ("mod", "mod") => {
```

Comment:
It think this is wrong. Previously you could do something like move `foo.rs` to `bar.rs` and the module would rename itself as appropriate. Here it will only do a rename for `mod.rs` files.

Reply:
Thanks! Fixed now.